geode-core/src/main/java/org/apache/geode/internal/cache/AbstractUpdateOperation.java

```diff
@@ -144,16 +144,21 @@ public static boolean doPutOrCreate(LocalRegion rgn, EntryEventImpl ev, long las
           br.getPartitionedRegion().getPrStats().startApplyReplication();
         }
         try {
-          // if the oldValue is the DESTROYED token and overwrite is disallowed,
-          // then basicPut will set the blockedDestroyed flag in the event
-          boolean overwriteDestroyed = ev.getOperation().isCreate();
-          if (rgn.basicUpdate(ev, true /* ifNew */, false/* ifOld */, lastMod,
-              overwriteDestroyed)) {
-            rgn.getCachePerfStats().endPut(startPut, ev.isOriginRemote());
-            // we did a create, or replayed a create event
-            doUpdate = false;
-            updated = true;
-          } else { // already exists. If it was blocked by the DESTROYED token, then
+          if (rgn.getRegionEntry(ev.getKey()) == null) {
+            // if the oldValue is the DESTROYED token and overwrite is disallowed,
+            // then basicPut will set the blockedDestroyed flag in the event
+            boolean overwriteDestroyed = ev.getOperation().isCreate();
+            if (rgn.basicUpdate(ev, true /* ifNew */, false/* ifOld */, lastMod,
+                overwriteDestroyed)) {
+              rgn.getCachePerfStats().endPut(startPut, ev.isOriginRemote());
+              // we did a create, or replayed a create event
+              doUpdate = false;
+              updated = true;
+            } else { // already exists. If it was blocked by the DESTROYED token, then
+              // do no update.
+              doUpdate = checkIfToUpdateAfterCreateFailed(rgn, ev);
+            }
+          } else {// already exists. If it was blocked by the DESTROYED token, then
             // do no update.
             doUpdate = checkIfToUpdateAfterCreateFailed(rgn, ev);
```

Comment:
Another problem with this fix is that it now calls getRegionEntry and if it returns non-null it then calls checkifToUpdateAfterCreate. But checkIfToUpdateAfterCreate is counting on basicUpdate having already been called which is the code that will set oldValueIsDestroyedToken() and isConcurrencyConflict() on "ev" which is all that checkIfToUpdateAfterCreate looks at.
I think the old code was okay because it basically said that if it found a DestroyedToken or had a concurrency conflict on the first call of basicUpdate then it is done and a second call does not need to happen.